packages/sit-onyx/src/components/OnyxBadge/OnyxBadge.vue

```diff
@@ -0,0 +1,94 @@
+<script lang="ts" setup>
+import { useDensity } from "../../composables/density";
+import type { OnyxBadgeProps } from "./types";
+
+const props = withDefaults(defineProps<OnyxBadgeProps>(), {
+  variation: "primary",
+});
+
+const { densityClass } = useDensity(props);
+
+defineSlots<{
+  /**
+   * Badge content.
+   */
+  default(): unknown;
+}>();
+</script>
+
+<template>
+  <div class="onyx-badge" :class="[`onyx-badge--${props.variation}`, densityClass]">
+    <div class="onyx-badge--content onyx-truncation-ellipsis">
```

Comment:
actually, could you try to make the dom tree flat and combine those two? not sure if both layers are needed
```suggestion
  <div class="onyx-badge" :class="[`onyx-badge--${props.variation}`, densityClass, 'onyx-truncation-ellipsis']">
```

Reply:
I made it with two layers so I can set the max-width on the label to align with figma requirements